internal/provisioner/helm/controllers/bundledeployment_controller.go

```diff
@@ -297,11 +334,17 @@ func (r *BundleDeploymentReconciler) getReleaseState(cl helmclient.ActionInterfa
 	desiredRelease, err := cl.Upgrade(obj.GetName(), r.ReleaseNamespace, chrt, values, func(upgrade *action.Upgrade) error {
 		upgrade.DryRun = true
 		return nil
-	})
+	},
+		func(upgrade *action.Upgrade) error {
+			post.cascade = upgrade.PostRenderer
+			upgrade.PostRenderer = post
+			return nil
+		})
 	if err != nil {
 		return currentRelease, stateError, err
 	}
 	if desiredRelease.Manifest != currentRelease.Manifest ||
+		!reflect.DeepEqual(desiredRelease.Chart.Metadata, currentRelease.Chart.Metadata) ||
```

Comment:
Any idea on why this comparison is needed?

Reply:
This is checking for the changes in Chart.yaml file.  It's contents are [here](https://pkg.go.dev/helm.sh/helm/v3@v3.9.4/pkg/chart#Metadata).